src/runtime-rs/crates/resource/src/network/utils/link/manager.rs

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use netlink_packet_route::{
+    link::nlas::{Info, InfoBridge, InfoData, InfoKind, Nla},
+    LinkMessage,
+};
+
+use super::{Link, LinkAttrs};
+
+pub fn get_link_from_message(mut msg: LinkMessage) -> Box<dyn Link> {
+    let mut base = LinkAttrs {
+        index: msg.header.index,
+        flags: msg.header.flags,
+        link_layer_type: msg.header.link_layer_type,
+        ..Default::default()
+    };
+    if msg.header.flags & libc::IFF_PROMISC as u32 != 0 {
+        base.promisc = 1;
+    }
+    let mut link: Option<Box<dyn Link>> = None;
+    while let Some(attr) = msg.nlas.pop() {
+        match attr {
+            Nla::Info(infos) => {
+                link = Some(link_info(infos));
+            }
+            Nla::Address(a) => {
+                base.hardware_addr = a;
+            }
+            Nla::IfName(i) => {
+                base.name = i;
+            }
+            Nla::Mtu(m) => {
+                base.mtu = m;
+            }
+            Nla::Link(l) => {
+                base.parent_index = l;
+            }
+            Nla::Master(m) => {
+                base.master_index = m;
+            }
+            Nla::TxQueueLen(t) => {
+                base.txq_len = t;
+            }
+            Nla::IfAlias(a) => {
+                base.alias = a;
+            }
+            Nla::Stats(_s) => {}
+            Nla::Stats64(_s) => {}
+            Nla::Xdp(_x) => {}
+            Nla::ProtoInfo(_) => {}
+            Nla::OperState(_) => {}
+            Nla::NetnsId(n) => {
+                base.net_ns_id = n;
+            }
+            Nla::GsoMaxSize(i) => {
+                base.gso_max_size = i;
+            }
+            Nla::GsoMaxSegs(e) => {
+                base.gso_max_seqs = e;
+            }
+            Nla::VfInfoList(_) => {}
+            Nla::NumTxQueues(t) => {
+                base.num_tx_queues = t;
+            }
+            Nla::NumRxQueues(r) => {
+                base.num_rx_queues = r;
+            }
+            Nla::Group(g) => {
+                base.group = g;
+            }
+            _ => {}
```

Comment:
There is no logging available in this module but I wonder if we should display a debug message if we hit this so we can tell if new attributes get added?

Reply:
Thanks for your review, add log has too many messages, add a comment.